chia/wallet/nft_wallet/nft_puzzles.py

```diff
@@ -255,14 +254,18 @@ def get_metadata_and_phs(unft: UncurriedNFT, solution: SerializedProgram) -> Tup
             # metadata update
             metadata = update_metadata(metadata, condition)
             metadata = Program.to(metadata)
-        elif condition_code == 51 and int_from_bytes(condition.rest().rest().first().atom) == 1:
-            # destination puzhash
-            if puzhash_for_derivation is not None:
-                # ignore duplicated create coin conditions
-                continue
-            memo = bytes32(condition.as_python()[-1][0])
-            puzhash_for_derivation = memo
-            log.debug("Got back puzhash from solution: %s", puzhash_for_derivation)
+        elif condition_code == 51:
+            atom = condition.rest().rest().first().atom
+            assert atom is not None
+
+            if int_from_bytes(atom) == 1:
```

Comment:
Could we do `as_int()` here, and that will throw a more helpful exception if necessary.

Reply:
Yeah we could also save an indentation layer by moving the inside if back to an `and` but I'll resolve and leave that optional for you.